src/GitHub.App/Services/GitClient.cs

```diff
@@ -424,16 +424,7 @@ public Task<bool> IsHeadPushed(IRepository repo)
 
             return Task.Factory.StartNew(() =>
             {
-                if (repo.Head.IsTracking)
-                {
-                    var trackedBranchTip = repo.Head.TrackedBranch.Tip;
-                    if (trackedBranchTip != null)
-                    {
-                        return repo.Head.Tip.Sha == trackedBranchTip.Sha;
-                    }
-                }
-
-                return false;
+                return repo.Head.TrackingDetails.AheadBy == 0;
```

Comment:
If you look at the documentation for [`TrackingDetails.AheadBy`](https://github.com/libgit2/libgit2sharp/blob/master/LibGit2Sharp/BranchTrackingDetails.cs#L30-L31) it says:

> This property will return <c>null</c> if this local branch has no upstream configuration or if the upstream branch does not exist

`TrackingDetails.AheadBy` will be null if there's no remote branch, and thus `== 0` will return `false`. `TrackingDetails` itself [will not be null]( https://github.com/libgit2/libgit2sharp/blob/9a817484df4a4025fa0b004c5718350f63f1a522/LibGit2Sharp/Branch.cs#L98).


So I believe this code is correct.